broker.go

```diff
@@ -675,69 +716,86 @@ func (b *Broker) muCloseDeadConnection(conn *connection) {
 // offset will return offset value for given partition. Use timems to specify
 // which offset value should be returned.
 func (b *Broker) offset(topic string, partition int32, timems int64) (offset int64, err error) {
-	conn, err := b.muLeaderConnection(topic, partition)
-	if err != nil {
-		return 0, err
-	}
-	resp, err := conn.Offset(&proto.OffsetReq{
-		ClientID:  b.conf.ClientID,
-		ReplicaID: -1, // any client
-		Topics: []proto.OffsetReqTopic{
-			{
-				Name: topic,
-				Partitions: []proto.OffsetReqPartition{
-					{
-						ID:         partition,
-						TimeMs:     timems,
-						MaxOffsets: 2,
+	var resErr error
+	for retry := 0; retry < b.conf.RetryErrLimit; retry++ {
```

Comment:
I agree that current approach is weird and it would be better to keep this in one place.  But I have no idea what the right level abstraction for this task. But It's definitely not a connection level. The main reason for this fix is a situation when one of the nodes is down - it this case a connection is broken and there is no reason to continue repeating - we should drop it and create a new one.